pkg/operator/rules.go

```diff
@@ -126,9 +127,83 @@ func (r *rulesReconciler) Reconcile(ctx context.Context, req reconcile.Request)
 	if err := r.ensureRuleConfigs(ctx, projectID, location, cluster); err != nil {
 		return reconcile.Result{}, fmt.Errorf("ensure rule configmaps: %w", err)
 	}
+
+	if err := r.scaleRuleConsumers(ctx); err != nil {
+		return reconcile.Result{}, fmt.Errorf("scale rule consumers: %w", err)
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *rulesReconciler) scaleRuleConsumers(ctx context.Context) error {
+	var desiredReplicas int32
+
+	var hasAnyRules bool
+	for _, check := range []RuleCheck{r.hasRules, r.hasClusterRules, r.hasGlobalRules} {
+		hasRules, err := check(ctx)
+		if err != nil {
+			return err
+		}
+		if hasRules {
+			hasAnyRules = true
+			break
+		}
+	}
+	if hasAnyRules {
+		desiredReplicas = 1
+	}
+
+	var alertManagerStatefulSet appsv1.StatefulSet
+	if err := r.client.Get(ctx, client.ObjectKey{Namespace: r.opts.OperatorNamespace, Name: "alertmanager"}, &alertManagerStatefulSet); err != nil {
+		return err
+	}
+	var ruleEvaluatorDeployment appsv1.Deployment
+	if err := r.client.Get(ctx, client.ObjectKey{Namespace: r.opts.OperatorNamespace, Name: "rule-evaluator"}, &ruleEvaluatorDeployment); err != nil {
+		return err
+	}
+
+	if *alertManagerStatefulSet.Spec.Replicas != desiredReplicas {
```

Comment:
Should we check if any alerts are configured in the rule-evaluator, if no alerts then we can turn off the alertmanager?

https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/pkg/operator/apis/monitoring/v1/types.go#L1356-L1359

Reply:
We could, if you think it's worthwhile. I was making the (perhaps incorrect) assumption that customers are likely to set up recording rules and alerting rules in the same place, whether that's in-cluster or externally. And it somewhat simplifies the solution to not have to inspect the rules for their type.

Given that the goal is just having both off by default, I'm not sure there's meaningful value added by handling them separately. Thoughts?

Reply:
I think it's out of scope for this PR, and might be overkill but I think there is no reason to have an alertmanager if you have no alerts. 

You are right alerts and recording rules are set in the same place. 

We can revist this later. perhaps addd a comment saying that could be possible. Up to you!